packages/core/rsc/stream-ui/stream-ui.ui.test.tsx

```diff
@@ -0,0 +1,217 @@
+const mockToolModel = new MockLanguageModelV1({
+  doStream: async () => {
+    return {
+      stream: convertArrayToReadableStream([
+        {
+          type: 'tool-call',
+          toolCallType: 'function',
+          toolCallId: 'call-1',
+          toolName: 'get_current_weather',
+          args: `{}`,
+        },
+        {
+          type: 'finish',
+          finishReason: 'stop',
+          logprobs: undefined,
+          usage: { completionTokens: 10, promptTokens: 3 },
+        },
+      ]),
+      rawCall: { rawPrompt: 'prompt', rawSettings: {} },
+    };
+  },
+});
+
+describe('rsc - streamUI()', () => {
+  it('should emit React Nodes with sync streamUI function', async () => {
+    const ui = await streamUI({
+      model: mockToolModel,
+      messages: [],
+      tools: {
+        get_current_weather: {
+          description: 'Get the current weather',
+          parameters: z.object({}),
+          generate: () => {
+            return <div>Weather</div>;
+          },
+        },
+      },
+    });
+
+    const rendered = await simulateFlightServerRender(ui.value);
+    expect(rendered).toMatchSnapshot();
+  });
+
+  it('should emit React Nodes with async streamUI function', async () => {
+    const ui = await streamUI({
+      model: mockToolModel,
+      messages: [],
+      tools: {
+        get_current_weather: {
+          description: 'Get the current weather',
+          parameters: z.object({}),
+          generate: async () => {
+            await new Promise(resolve => setTimeout(resolve, 100));
+            return <div>Weather</div>;
+          },
+        },
+      },
+    });
+
+    const rendered = await simulateFlightServerRender(ui.value);
+    expect(rendered).toMatchSnapshot();
+  });
+
+  it('should emit React Nodes with generator streamUI function', async () => {
+    const ui = await streamUI({
+      model: mockToolModel,
+      messages: [],
+      tools: {
+        get_current_weather: {
+          description: 'Get the current weather',
+          parameters: z.object({}),
+          generate: async function* () {
+            yield <div>Loading...</div>;
+            await new Promise(resolve => setTimeout(resolve, 100));
+            return <div>Weather</div>;
+          },
+        },
+      },
+    });
+
+    const rendered = await simulateFlightServerRender(ui.value);
+    expect(rendered).toMatchSnapshot();
+  });
+});
+
+describe('rsc - streamUI() onFinish callback', () => {
+  let result: Parameters<
+    Required<Parameters<typeof streamUI>[0]>['onFinish']
+  >[0];
+
+  beforeEach(async () => {
+    const ui = await streamUI({
+      model: mockToolModel,
+      messages: [],
+      tools: {
+        get_current_weather: {
+          description: 'Get the current weather',
+          parameters: z.object({}),
+          generate: () => {
+            return 'Weather';
+          },
+        },
+      },
+      onFinish: async event => {
+        result = event as unknown as typeof result;
```

Comment:
why is `as unknown as typeof result` needed here?

Reply:
It was something I copy-pasted from the `stream-object.test.ts` I was adapting the `onFinish` test from. Doesn't appear necessary so I'll remove 👍🏻